gh-114803: Mention that `@dataclass` should not be applied on enums (#114891)
python/cpython: Doc/howto/enum.rst

    >>> Creature.DOG
    <Creature.DOG: size='medium', legs=4>

Use the :func:`~dataclasses.dataclass` argument ``repr=False``
to use the standard :func:`repr`.

.. versionchanged:: 3.12
   Only the dataclass fields are shown in the value area, not the dataclass'
   name.

.. note::

>    Adding :func:`~dataclasses.dataclass` decorator to :type:`Enum`

Reviewer: ```suggestion
   Adding :func:`~dataclasses.dataclass` decorator to :class:`Enum`
```

   and its subclasses is not supported. It will not raise any errors,

>    but it will produce very strange results at runtime::

Reviewer 2: ```suggestion
   but it will produce very strange results at runtime, such as members
   being equal to each other::
```

>       @dataclass  # don't do this: it does not make any sense
>       class Colors(Enum):
>          red = 'red'

Reviewer 2: ```suggestion
      >>> @dataclass                      # don't do this: it does not make any sense
      ... class Color(Enum):
      ...    RED = 1
      ...    GREEN = 2
      ...    BLUE = 3
      ...
      >>> Color.RED is Color.BLUE
      False
      >>> Color.RED == Color.BLUE
      True
```

Author: Do we *really* want to assert this behaviour that seems at least partially incorrect? 🤔

Reviewer 2: The `==` is definitely incorrect, and showing it helps explain why using `@dataclass` on an enum is a bad idea.  We can certainly add a `# this should fail` to the `==` line, though.  Also, if that `==` test ever does work correctly, I want to know.  :smiley:

Author: Done 👍
